packages/react/src/hooks/actions/useStateMutationAction.ts

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { Hub } from '@aws-amplify/core';
+
+import {
+  ACTIONS_CHANNEL,
+  ACTION_STATE_MUTATION_FINISHED,
+  ACTION_STATE_MUTATION_STARTED,
+} from './constants';
+
+type StateChangeEvent = React.ChangeEvent<
+  HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+>;
+
+type UseStateMutationAction = [string, (event: StateChangeEvent) => void];
+
+export const useStateMutationAction = (
+  initialState: string
+): UseStateMutationAction => {
+  const [state, setState] = React.useState(initialState);
+
+  const setNewState = React.useCallback((event: StateChangeEvent) => {
+    const newState = event.currentTarget.value;
```

Comment:
I was wondering how general the use cases of `useStateMutationAction` would be. From [the golden file](https://code.amazon.com/packages/AmplifyStudioCommon/blobs/d76a84faef43f6ad8ea0b36031d6a932864b8f66/--/component_golden_files/react/form.tsx#L19), it is limited to only be used on form fields, which means the mutable state will always be a `string`.

Reply:
Had a discussion with @reesscot, we should have codegen to handle the event part: 
```ts
const [userNameTextFieldValue, setUserNameTextFieldValue] = useStateMutationAction('vizsla');
const handleOnChange = (e) => {
 const value = e.target.value; // may need some kind of check if value exists
 setUserNameTextFieldValue(value)
}
```

Will update the golden file

Reply:
Discussed offline, but I think we should just have you accept a value T as input, and effectively just wrap the setState hook w/ the hub actions. This way it's easier for customers to use this in the same way they use `useState` today.